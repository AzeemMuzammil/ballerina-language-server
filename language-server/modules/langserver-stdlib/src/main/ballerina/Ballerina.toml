[package]
org = "ballerina"
name = "langserverstdlib"
version = "0.1.0"
