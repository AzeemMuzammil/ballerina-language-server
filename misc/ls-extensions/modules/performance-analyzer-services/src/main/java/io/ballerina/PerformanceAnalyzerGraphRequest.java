package io.ballerina;

import org.eclipse.lsp4j.Range;
import org.eclipse.lsp4j.TextDocumentIdentifier;

public class PerformanceAnalyzerGraphRequest {
    private TextDocumentIdentifier documentIdentifier;
    private Range range;
    private String choreoToken;
    private String choreoCookie;

    public TextDocumentIdentifier getDocumentIdentifier() {

        return documentIdentifier;
    }

    public void setDocumentIdentifier(TextDocumentIdentifier documentIdentifier) {

        this.documentIdentifier = documentIdentifier;
    }

    public Range getRange() {

        return range;
    }

    public void setRange(Range range) {

        this.range = range;
    }

    public String getChoreoToken() {

        return choreoToken;
    }

    public void setChoreoToken(String choreoToken) {

        this.choreoToken = choreoToken;
    }

    public String getChoreoCookie() {

        return choreoCookie;
    }

    public void setChoreoCookie(String choreoCookie) {

        this.choreoCookie = choreoCookie;
    }
}
