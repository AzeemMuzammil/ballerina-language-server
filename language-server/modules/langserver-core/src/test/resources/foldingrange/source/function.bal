import ballerina/io;
import ballerina/math;

public function main() {
    io:println("Hello, World!");
}

# Prints PI value
# Refers math library
public function printPI(int value) {
    // Refer symbols of another module.
    // `math:PI` is a qualified identifier. Note the usage of the module prefix.
    float piValue = math:PI;

    // Use the explicit prefix `console` to invoke a function defined in the `ballerina/io` module.
    io:println(piValue);
    if value == 0 {
        io:println("Value cannot be zero.");
    } else if value < 0 {
        io:println("Value cannot be negative.");
    } else {
        io:println("Value is acceptable.");
    }
}
