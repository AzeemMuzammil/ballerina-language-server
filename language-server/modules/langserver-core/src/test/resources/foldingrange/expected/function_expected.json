{
  "result":[
    {
      "startLine":3,
      "endLine":4,
      "kind":"region"
    },
    {
      "startLine":7,
      "endLine":8,
      "kind":"comment"
    },
    {
      "startLine":9,
      "endLine":22,
      "kind":"region"
    },
    {
      "startLine":16,
      "endLine":17,
      "kind":"region"
    },
    {
      "startLine":18,
      "endLine":19,
      "kind":"region"
    },
    {
      "startLine":20,
      "endLine":21,
      "kind":"region"
    },
    {
      "startLine":0,
      "endLine":1,
      "kind":"region"
    }
  ]
}
